sources/engine/Stride.Input/Windows/InputSourceWindowsDirectInput.cs

```diff
@@ -116,6 +118,44 @@ public void OpenDevice(DeviceInstance deviceInstance)
                 RegisterDevice(controller);
             }
         }
+
+        private bool IsXInputDevice(Guid productGuid)
+        {
+            // Set security level to IMPERSONATE
+
+            DComSessionOptions DComOptions = new DComSessionOptions();
+            DComOptions.Impersonation = ImpersonationType.Impersonate;
+
+            var mySession = CimSession.Create(null, DComOptions);
+
+            IEnumerable<CimInstance> allDevices = mySession.QueryInstances(@"root\cimv2", "WQL", "SELECT * FROM Win32_PNPEntity");
```

Comment:
I imagine this query is not super cheap and results won't be different if called more than once a frame. Consider calling the query and transforming the IEnumerable with regex once, caching the result in the `Scan()` method, so that `IsXInputDevice` can be a simpler lookup.

Reply:
@Jklawreszuk - do you think your WQL optimization is enough here? The `Scan()` method goes in a loop over the devices and checks for each one if it's XInput - are we confident to say there usually wouldn't be many devices returned by DirectInput and thus this isn't a big perf issue? Otherwise we make a Windows call a few times in a tight loop and I imagine those are usually slower. However, at the same time Scan is called on initialization and then only upon manual request it seems.

Reply:
@manio143 Yea, Very good suggestion 👍. I guess, It is unlikely that something will change under the scan() method call